# Fst and Fet evaluation of mpileups
## Merging separate aligned .bam files into two "I" and "P"
module load bioinfo-tools
module load samtools

samtools merge merged.I.bam -b list_bam_I.txt
samtools merge merged.P.bam -b list_bam_P.txt

## Generating mpileups
samtools mpileup -d 500 --min-MQ 30 --min-BQ 30 --adjust-MQ 50 -f /domus/h1/pauliusb/Haemonchus_2018_genome/haemonchusnewest.fa \
merged.I.bam \
merged.P.bam \
-o IP2.unchecked.mpileup
sed 's/\t\t/\t!\t!/g' IP2.unchecked.mpileup > IP2.final.mpileup

## Spliting mpileups by sample (I or P) and chr

## Syncing
module load bcftools
module load popoolation2
mpileup2sync.pl --input IP2.final.mpileup --min-qual 20 --output IP2_filtered.sync

## Finding +5 bp indels and removing them from the synced files
perl /sw/bioinfo/popoolation2/1201/rackham/indel_filtering/identify-indel-regions.pl --min-count 2 --indel-window 5 --input IP2_filtered.sync --output indels.gtf
perl /sw/bioinfo/popoolation2/1201/rackham/indel_filtering/filter-sync-by-gtf.pl --input IP2_filtered.sync --output IP_noindels.sync --gtf indels.gtf


